util/argo/resource_tracking.go

```diff
@@ -244,3 +245,20 @@ func (rt *resourceTracking) Normalize(config, live *unstructured.Unstructured, l
 
 	return nil
 }
+
+func (rt *resourceTracking) RemoveAppInstance(un *unstructured.Unstructured) error {
```

Comment:
Can you move this up to immediately after `SetAppInstance`, since their behavior should be very closely mirrored?